src/pages/filter/ui/gene-region.tsx

```diff
@@ -1,101 +1,120 @@
 import { PanelButtons } from './panelButtons'
 import { DisabledVariantsAmount } from './query-builder/ui/disabled-variants-amount'
 
-export interface IGeneRegionFormValues {
-  locus: string
+interface IGeneRegionCachedValues {
+  conditions: {
+    locus: string
+  }
 }
 
-export const GeneRegion = observer(
-  ({
-    values: { locus },
-    setFieldValue,
-    submitForm,
-    resetForm,
-  }: FormikProps<IGeneRegionFormValues>) => {
-    const cachedValues = filterStore.readFilterCondition<IGeneRegionFormValues>(
-      FuncStepTypesEnum.GeneRegion,
-    )
+export const GeneRegion = observer(() => {
+  const cachedValues = filterStore.readFilterCondition<IGeneRegionCachedValues>(
+    FuncStepTypesEnum.GeneRegion,
+  )
 
-    const locusValue = cachedValues?.locus || ''
-    const { variants } = filterStore.statFuncData
+  const locusValue: string = cachedValues?.conditions.locus || ''
+  const { variants } = filterStore.statFuncData
 
-    const [isErrorVisible, setIsErrorVisible] = useState(false)
+  const selectedFilterName: string = filterStore.selectedGroupItem.name
+  const selectedFilterGroup: string = filterStore.selectedGroupItem.vgroup
 
-    const validateValue = (value: string) => {
-      validateLocusCondition({ value, setIsErrorVisible })
-    }
+  const [isErrorVisible, setIsErrorVisible] = useState(false)
 
-    useEffect(() => {
-      const params = `{"locus":"${locusValue}"}`
+  const validateValue = (value: string) => {
+    validateLocusCondition({ value, setIsErrorVisible })
+  }
 
-      filterStore.fetchStatFuncAsync('GeneRegion', params)
+  useEffect(() => {
+    const params = `{"locus":"${locusValue}"}`
 
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [])
+    filterStore.fetchStatFuncAsync('GeneRegion', params)
 
-    useEffect(() => {
-      filterStore.setFilterCondition<IGeneRegionFormValues>(
+    return () => filterStore.resetStatFuncData()
+  }, [locusValue])
+
+  const setLocusValue = (value: string) => {
+    filterStore.setFilterCondition<IGeneRegionCachedValues>(
+      FuncStepTypesEnum.GeneRegion,
+      {
+        conditions: { locus: value },
+      },
+    )
+  }
+
+  const handleResetFields = () => {
+    filterStore.clearFilterCondition(FuncStepTypesEnum.GeneRegion)
+  }
+
+  const handleSumbitCondtionsAsync = async () => {
+    if (datasetStore.activePreset) datasetStore.resetActivePreset()
+
+    await datasetStore.setConditionsAsync([
+      [
+        'func',
         FuncStepTypesEnum.GeneRegion,
-        {
-          locus,
-        },
-      )
-    }, [locus])
-
-    const handleResetFields = () => {
-      filterStore.clearFilterCondition(FuncStepTypesEnum.GeneRegion)
+        '',
+        ['True'],
+        { locus: locusValue },
+      ],
+    ])
+
+    filterStore.addSelectedFilters({
+      group: selectedFilterGroup,
+      groupItemName: selectedFilterName,
+      variant: [`{"locus":"${locusValue}"}`, 0],
+    })
```

Comment:
create /filter/ui/panels/function-panel/components/gene-region/gene-region.store.ts and you can move handlers and states there. It separates view component and business logic. And also smaller chunks of code are easier to understand. You can look up how it's done in [forome-565 PR](https://github.com/ForomePlatform/Anfisa-React-Client/pull/412/files).